fbs/matrix.fbs

```diff
@@ -22,39 +22,70 @@
 
 namespace NetEncoding;
 
-table Float32Array {
+table Float32FBArray {
   data: [float32];
 }
 
-table Uint32Array {
+table Uint32FBArray {
   data: [uint32];
 }
 
-table Int32Array {
+table Int32FBArray {
   data: [int32];
 }
 
-table Float64Array {
+table Float64FBArray {
   data: [float64];
 }
 
-table JSONEncodedArray {
+table JSONEncodedFBArray {
   // contains a UTF-8/JSON encoded array.  Used to store other
   // types (or polymorphic arrays)
   data: [uint8];
 }
 
-union TypedArray {
-  Float32Array,
-  Int32Array,
-  Uint32Array,
-  Float64Array,
-  JSONEncodedArray
+table CatInt8FBArray {
+  data: [int8];
+  codes: [uint8];
+}
+
+table CatInt16FBArray {
+  data: [int16];
+  codes: [uint8];
+}
+
+table CatInt32FBArray {
+  data: [int32];
+  codes: [uint8];
+}
+
+table SparseFloat32FBArray {
```

Comment:
please add comment describing data structure.  I can't quite discern if this is a COO matrix, or something else.